File: ts/metrics/dimension.py


"""
Dimension class for model server metrics
"""


class Dimension(object):
    """
    Dimension class defining key value pair
    """
    def __init__(self, name, value):
        """
        Constructor for Dimension class

        Parameters
        ----------
        name: str
            NAme of dimension
        value : str
           Unique Value of dimension
        """
        self.name = name
        self.value = value

    def __str__(self):
        """
        Return a string value
        :return:
        """
        return "{}:{}".format(self.name, self.value)

    def to_dict(self):
        """
        return an dictionary
        """
        return {'Name': self.name, 'Value': self.value}

File: cpp/src/backends/metrics/dimension.cc
#include "src/backends/metrics/dimension.hh"

namespace torchserve {
    Dimension::Dimension(const std::string& name, const std::string& value) :
        name{name}, value{value} {}

    std::string Dimension::ToString() const {
        return name + ":" + value;
    }

} // namespace torchserve
